### ISSUE TITLE
Harbor Chart, notes.txt error during upgrade with external database

### ISSUE THREAD

**Issue by mkuendig** [bitnami/charts#3678]
**Which chart**:
bitnami/harbor

**Describe the bug**
while doing a helm upgrade -i on an exiting chart I get missing password errors when I use an external postgres. This is not logical as the passwords are coded into the helm chart and second replication is handled externally.

**To Reproduce**

1. helm upgrade -i harbor bitnami/harbor
2. change something in the values.yaml file
3. helm upgrade -i harbor bitnami/harbor

**Expected behavior**
Upgrade succeded. Now it gives an error in the displaying of the notes.txt file.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

3.0.2

- Output of `kubectl version`:

1.17

**Additional context**

Some logs:
```
U:\k8_stuff\apps\harbor>helm upgrade -i harbor-portal-ut bitnami\harbor\ -n harbor-portal-ut -f bitnami\harbor\values-production.yaml -f eng-values.yaml
Error: UPGRADE FAILED: template: harbor/templates/NOTES.txt:66:4: executing "harbor/templates/NOTES.txt" at <include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordErrors "context" $)>: error calling include: template: harbor/charts/common/templates/_errors.tpl:18:48: executing "common.errors.upgrade.passwords.empty" at <fail>: error calling fail:
PASSWORDS ERROR: you must provide your current passwords when upgrade the release
    'postgresql.replication.password' must not be empty, please add '--set postgresql.replication.password=$POSTGRESQL_REPLICATION_PASSWORD' to the command. To get the current value:
 
        export POSTGRESQL_REPLICATION_PASSWORD=$(kubectl get secret --namespace harbor-portal-ut harbor-portal-ut-postgresql -o jsonpath="{.data.postgresql-replication-password}" | base64 --decode)
```

**Comment by rafariossaa** [comment]
Hi,
  If the passwords are not set, they are randomly generated by helm each time it is run.  
  You can check the code [here](https://github.com/bitnami/charts/blob/7ccff309e181b6c1e1766a6d6cf59a6c6c13deda/bitnami/harbor/templates/core/core-secret-envvars.yaml#L26):
```...
  {{- if .Values.harborAdminPassword }}
  HARBOR_ADMIN_PASSWORD: {{ .Values.harborAdminPassword | b64enc | quote }}
  {{- else if (not .Values.forcePassword) }}
  HARBOR_ADMIN_PASSWORD: {{ randAlphaNum 10 | b64enc | quote }}
```
  So as you are not providing the old password helm doesn't know about the old one. 
  Have into account that helm is basically a template renderer.

**Comment by mkuendig** [comment]
@rafariossaa Thanks for the answer. However, I believe that the error is not related to the Harbor_Admin_Password. It says it is related to Postgres.replication password as indicated in this line:

postgresql.replication.password' must not be empty, please add '--set postgresql.replication.password=$POSTGRESQL_REPLICATION_PASSWORD' to the command. To get the current value

As Postgres is now external, this error is wrong in my opinion.

**Comment by rafariossaa** [comment]
Hi,
  As it is external, did you provide the password ? When you run a second time helm it knows nothing about the paramters you set the first time.

**Comment by mkuendig** [comment]
yes, the DB password is set, else it wouldn't know how to connect to the database.

in this case we are not using replication. So no replication password is set, but as that anyway would be external it is not relevant for helm or harbor.

**Comment by carrodher** [comment]
Hi, thanks for reporting this issue, it was due to an error introduced in the common subchart, we already released a new version of the subchart fixing this issue (see https://github.com/bitnami/charts/pull/3733).

Now we are working on releasing a new version of the Harbor chart (there is a new version in the upstream project itself, Harbor 2.1.0). Once released this new version the issue should be solved. As the new version contains several changes we need to do some changes on our side, it is not released automatically, that is why we are taking more time than a regular release.
